src/Handlers/PaymentResponseHandler.php

```diff
@@ -67,120 +88,197 @@ public function __construct(
-
                 break;
-            case 'RedirectShopper':
-            case 'IdentifyShopper':
-            case 'ChallengeShopper':
-                // Store response for cart temporarily until the payment is done
+            case self::REDIRECT_SHOPPER:
+            case self::IDENTIFY_SHOPPER:
+            case self::CHALLENGE_SHOPPER:
+            case self::RECEIVED:
+            case self::PRESENT_TO_SHOPPER:
+                // Store response for cart until the payment is finalised
                 $this->paymentResponseService->insertPaymentResponse(
                     $response,
-                    $transaction->getOrder()->getOrderNumber(),
+                    $orderNumber,
                     $salesChannelContext->getToken()
                 );
-                return new RedirectResponse('responseUrl');
-                break;
-            // Received and PresentToShopper follow the same protocol at this stage
-            case 'Received':
-            case 'PresentToShopper':
-                // Store payments response for later use
-                $context = $salesChannelContext->getContext();
-                // Return to frontend with additionalData or action
-                $customFields = array_merge(
-                    $transaction->getOrderTransaction()->getCustomFields() ?: [],
-                    ['additionalData' => $response['additionalData']]
-                );
-                $this->orderTransactionRepository->update(
-                    ['id' => $orderTransactionId, 'customFields' => $customFields],
-                    $context
-                );
-                // Tag the order as waiting for payment
-                $this->transactionStateHandler->process($orderTransactionId, $context);
+
+                // Return the standard payment response handler result
+                return $this->paymentResponseHandlerResult;
                 break;
-            case 'Error':
+            case self::ERROR:
                 // Log error
                 $this->logger->error(
-                    "There was an error with the payment method. id:  " . $orderTransactionId .
+                    'There was an error with the payment method. ' .
                     ' Result code "Error" in response: ' . print_r($response, true)
                 );
-                // Cancel the order
-                throw new AsyncPaymentProcessException(
-                    $orderTransactionId,
-                    'The payment had an error'
-                );
+
                 break;
             default:
                 // Unsupported resultCode
                 $this->logger->error(
-                    "There was an error with the payment method. id:  " . $orderTransactionId .
+                    "There was an error with the payment method. id:  " .
                     ' Unsupported result code in response: ' . print_r($response, true)
                 );
+        }
+    }
+
+    public function handleShopwareAPIs(
```

Comment:
Just for consistency with the second function
```suggestion
    public function handleShopwareApis(
```